### PR TITLE
Fixes AI eyes able to be shuttlecrushed

### LINKED ISSUE #23084: The ai eye can be shuttle gibbed.
**Issue Description**:

Yeah, i dunno. it can be shuttle gibbed for whatever reason, deleting it, making the ai largely not functional aside from what it can see within the range of its core

**What did you expect to happen**: 

The eye to NOT be shuttle gibbed

**What happened instead**:

https://github.com/ParadiseSS13/Paradise/assets/96908085/83102aa1-d361-4b0c-90f1-ec4e0c7b9210

You're unable to move after your view is reset to the core.

**Why is this bad/What are the consequences**:

see video

**Steps to reproduce the problem**:

1. be ai
2. have your eye where a shuttle lands
3. have your eye die.
4. require carding to be fixed

**When did the problem start happening**:

Unsure, best bet would be pr #22935?

**Extra information**:

Apparently the same happens with advanced cam consoles from what coding chat is saying.

### PR DESCRIPTION
**Pull request by DGamerL**
## What Does This PR Do
Fixes #23084 

## Why It's Good For The Game
A non-physical representation of an Eye should not be able to be squashed like a bug

## Testing
Didn't get squashed like a bug under the shuttle, instead nothing happened

## Changelog
:cl:
fix: Fixed AI eyes able to be shuttlecrushed
/:cl:

### CODE REVIEW

**Comment by Screampuff** [comment]
Holy shit

**Review by Henri215 — APPROVED** [review]

**Review by Sirryan2002 — APPROVED** [review]

File: code/modules/mob/living/silicon/ai/freelook/eye.dm [hunk 1]
```diff
@@ -10,6 +10,7 @@
 	icon_state = "eye"
 	alpha = 127
 	invisibility = SEE_INVISIBLE_OBSERVER
+	flags_2 = IMMUNE_TO_SHUTTLECRUSH_2
```
Comment (Contrabang) [inline code/modules/mob/living/silicon/ai/freelook/eye.dm:13]: If you dont mind, all mob cameras should probably be immune to shuttlecrushing

**Review by Contrabang — CHANGES_REQUESTED** [review]
Small request, otherwise LGTM

**Review by Contrabang — APPROVED** [review]
LGTM, no subtypes of /mob/camera override flags_2 from what I can see

**Review by farie82 — APPROVED** [review]

**Pull request merged**